osa/scripts/calibration_pipeline.py

```diff
@@ -157,9 +156,19 @@ def calibrate_charge(
     cmd = calibration_file_command(
         drs4_pedestal_run_id=drs4_pedestal_run_id, pedcal_run_id=pedcal_run_id
     )
-    analysis_step = ChargeCalibrationStage(run=f"{pedcal_run_id:05d}", command_args=cmd)
-    analysis_step.execute()
-    return analysis_step.rc
+    
+    try:
+        analysis_step = ChargeCalibrationStage(run=f"{pedcal_run_id:05d}", command_args=cmd)
+        analysis_step.execute()
```

Comment:
I wonder how this works with the `.execute()` method, which retries n times if rc != 0. Maybe the retry with filter=52 could be done inside that method of the general `AnalysisStage`. Although I think it would work similarly to your implementation.

Also, have you checked whether we need some kind of cleanup before the retry?

Reply:
In principle I have tested it like this and it works. The `.execute() `method of the `ChargeCalibrationStage` doesn't retry it, it just raises an error, so if` rc!=0`, it will just execute the code under `except`: 
```
def execute(self):
        """Run the program and retry if it fails."""
        log.info(f"Executing {stringify(self.command_args)}")
        output = sp.run(self.command_args, stdout=sp.PIPE, stderr=sp.STDOUT, encoding="utf-8")
        self.rc = output.returncode
        self._write_checkpoint()

        # If fails, remove products from the directory for subsequent trials
        if self.rc != 0:
            self._clean_up()
            raise ValueError(f"{self.command} failed with output: \n {output.stdout}")
```